### ISSUE TITLE
Colab tutorial on "how to fine-tune language models such as USE"

### ISSUE THREAD

**Issue by AmmarRashed** [tensorflow/hub#605]
How to fine-tune LMs for supervised and unsupervised tasks?
How to do resume training of something like USE (i.e. adoption training) on domain specific corpus?
A colab tutorial for that would be greatly appreciated.

Thanks,

**Comment by s4sarath** [comment]
Much appreciated :-)

**Comment by maziyarpanahi** [comment]
I am also very interested in this. I've followed the other issue which resulted in this feature request. I agree with the statements made as of why the pre-trained models such as USE should be fine-tuned during the training on a specific task, but I also believe they should be able to be fine-tuned and be resaved to be served in domain-specific tasks later on. 

I am looking forward to seeing end-to-end fine-tuning LM models such as USE or BERT and save them back as SavedModel/Hub Module so we can load them later.

**Comment by akhorlin** [comment]
The discussion on https://github.com/tensorflow/hub/issues/641 might have useful points to how resaving USE can be accomplished.

**Comment by anirbankonar123** [comment]
Hi, we are interested to fine-tune the embedding in un-supervised way. We do not have labels for Classification task. Thanks

**Comment by akhorlin** [comment]
You could try to construct tasks from an supervised dataset similar to how it is done for BERT, e.g. Masked Language Model task (https://github.com/tensorflow/models/blob/master/official/nlp/modeling/layers/masked_lm.py)

**Comment by WGierke** [comment]
We now have a Colab tutorial on [fine-tuning BERT](https://www.tensorflow.org/official_models/fine_tuning_bert), which also saves and re-loads the model in the end. Hope this helps!

**Comment by AmmarRashed** [comment]
Thanks a lot for the support. The documentation looks great!

**Comment by jjccooooll12** [comment]
Hi, do we have any news on this? I see that the issue has been closed but it doesn't look like we have a way to finetune the LM of the USE

**Comment by anirbankonar123** [comment]
I think we need to shift to Transformers

**Comment by s4sarath** [comment]
Check tf-transformers.
 github.com/legacyai/tf-transformers
